### DISCUSSION TITLE
关于环境变量配置，新用umi，有几个问题

### DISCUSSION THREAD

**Discussion by iHaroro** [umijs/umi#8304] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/22785090/176362796-46c6841c-cd4d-461c-a445-f2066a94284c.png)
👆🏻这是我的配置相关
其中FOO变量可以使用，但是我的TS却在告诉我error，但是实际控制台是有bar的字符串打印的
```javascript
console.log(FOO) // 'bar'
TS2304: Cannot find name 'FOO'.
```
然后使用了这种方式防止TS报错，但是并未取到值
```javascript
console.log((window as any).FOO) // undefined
```
question：这个FOO的值该怎么取？

**Comment by fz6m** [comment, marked as answer]
create file `./src/global.d.ts` :

```ts
// ./src/global.d.ts

declare global {
    const FOO: string
}

export {}
```

**Reply by iHaroro** [reply]
我在 ``project-root/src/typings.d.ts``中声明了对应的环境变量解决了问题，感谢提供的思路🙏

**Comment by xiaohuoni** [comment]
// @ts-ignore
console.log(FOO) // 'bar
